[iuhiah] iP

iuhiah opened the pull request:
# About Bibo!

### Who Bibo?

Can't seem to remember what's going on in your life? Try asking Bibo for help! Save your todos, tasks and events in the blink of an eye by telling Bibo:

* what you need to do
* when you need to do it
* or if you don't have to do it anymore (yay!)


### Why Bibo?

Bibo is capable of:

- [x] Remembering your tasks
- [x] Showing your task list at a glance
- [x] Saying goodbye \:(

*Full list of features can be found [here](https://github.com/iuhiah/ip?tab=readme-ov-file#features)!*

> "Beep boop... Beeboo... Bibo?" - Bibo, probably

### Use Bibo?
If you can type and understand English, there's nothing stopping you from:

1. Downloading Bibo from [here](https://github.com/iuhiah/ip/releases/download/v1.0.0/bibo-1.0.0.jar)
2. Following the installation instructions [here](https://github.com/iuhiah/ip?tab=readme-ov-file#installation)
3. Entrusting your life to Bibo 💕 (Bibo loves you too)

### How Bibo?

If you wish to interact with the source code, you can clone the repository by running:
```bash
git clone https://github.com/iuhiah/ip.git
```
The main method can be found in `src/main/java/Bibo.java`.

---

Have a great day (with Bibo)! 🌟

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

iuhiah committed:
Project setup: rename Duke to Bibo, update README, add .class file exclusion

iuhiah committed:
Echo user input

iuhiah committed:
Add to-do list

iuhiah committed:
Add Task class, message output cleanup

iuhiah committed:
Add Task subtypes: Todo, Deadline, Event. Extract Bibo actions into functions.

iuhiah committed:
Updated test scripts + grammar fix

iuhiah committed:
Improve error handling

iuhiah committed:
More specific error handling, introduce command enums

iuhiah committed:
Implement ArrayList, add delete function

iuhiah committed:
Refactor project structure: Create bibo.task and bibo.exception packages, improve error handling.

iuhiah committed:
Extract handling of user input to Parser.java. Update test scripts and related files.

iuhiah committed:
Add file handling and improve error handling

iuhiah committed:
Merge Level 7 branch to master

iuhiah committed:
Add handing for LocalDateTime objects

iuhiah committed:
Merge Level 8 branch to master

iuhiah committed:
Add comment headers to exceptions and update test related files

iuhiah committed:
Refactor project structure: extract Ui and TaskList classes

iuhiah committed:
Fix styling issues

iuhiah committed:
Add gradle support

iuhiah committed:
Fix formatting issues and add Checkstyle support

iuhiah committed:
Extract Command enums, fix styling issues

iuhiah committed:
Add tests for Parser class (dummy commit to fulfil branch requirement)

iuhiah committed:
Update tests for Storage class (dummy commit for branch requirement)

iuhiah committed:
Merge branch 'branch-A-javadoc'

iuhiah committed:
Merge branch 'branch-A-CodingStandard'

iuhiah committed:
Add JAR packaging to build file, fix styling issues

iuhiah committed:
Rename org folder to iuhiah

iuhiah committed:
Fix Shadow attributes in build file, add installation notes

iuhiah committed:
Fix Shadow attributes in build file, add installation notes

iuhiah committed:
Update installation instructions in README

iuhiah committed:
Merge branch 'master' of https://github.com/iuhiah/ip

iuhiah committed:
Add basic JavaFX GUI

iuhiah committed:
Merge branch 'Level-10'

GauhVish commented on src/main/java/bibo/Command.java:
```diff
@@ -0,0 +1,178 @@
+            }
+            messages.clear();
+        }
+
+        protected String getResponse() {
+            StringBuilder message = new StringBuilder("");
+            for (String msg : messages) {
+                message.append(msg).append("\n");
+            }
+            messages.clear();
+            return message.toString();
+        }
+    }
```
Hi, i think though this is a getter function, its sighly more complicated than your average getter type function, i think it should have a javadocs : )

GauhVish commented on src/main/java/bibo/Command.java:
```diff
@@ -0,0 +1,178 @@
+            return message.toString();
+        }
+    }
+
+    protected void run(String args, TaskList taskList) {
+        try {
+            cmd.execute(args, taskList);
+            cmd.printMessages();
+            storage.saveTaskList(taskList);
+        } catch (BiboException e) {
+            ui.speak(e.getMessage());
+        }
+    }
+
+    protected String getResponse(String args, TaskList taskList) {
+        try {
+            cmd.execute(args, taskList);
+            storage.saveTaskList(taskList);
+        } catch (BiboException e) {
+            return e.getMessage();
+        }
+
+        return cmd.getResponse();
+    }
+}
```
Same for these 2, seem to missing their respective javadocs

GauhVish commented on src/main/java/bibo/controllers/MainWindow.java:
```diff
@@ -0,0 +1,74 @@
+    private Image biboImage = new Image(this.getClass().getResourceAsStream("/images/bibo.png"));
+
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+        String greeting = "Beep boop! Bibo!";
+        dialogContainer.getChildren().add(
+                DialogBox.getBiboDialog(greeting, biboImage)
+        );
+    }
```
I would suggest adding a javadocs here as well : )

GauhVish commented on src/main/java/bibo/exceptions/TaskFormatException.java:
```diff
@@ -0,0 +1,57 @@
+    public String toString() {
+        return "TaskDescriptionException: " + getMessage();
+    }
+}
```
They way you handle the code with enums is really nice : ), it makes the code look super good, (even in the other files not just this one)

iuhiah replied:
Thanks for the comments, I'll work through them and add JavaDocs to the methods mentioned :) Just wanted to add that the phrasing of the overall comment really made my day!

GauhVish commented on src/main/java/bibo/utils/FileParser.java:
```diff
@@ -0,0 +1,51 @@
+
+            taskType = splitTaskData[0];
+            switch (taskType) {
+            case "T":
+                taskType = "todo";
+                break;
+            case "D":
+                taskType = "deadline";
+                splitTaskData[2] = splitTaskData[2]
+                        .replace(" (by: ", " /by ")
+                        .replace(")", "");
+                break;
+            case "E":
+                taskType = "event";
+                splitTaskData[2] = splitTaskData[2]
+                        .replace(" (from: ", " /from ")
+                        .replace(" to ", " /to ")
```
Good job using the correct switch case formatting !

GauhVish replied:
Throughout all the files : )

GauhVish reviewed (commented):
Wow this is amazingly neat and clean . I genuinely could not find something that was wrong with code standard, even the most simple variables were all appropriately handled. I only suggested adding some javadocs for some functions that went a bit beyond what traditional getter/setters would do : )

iuhiah committed:
Fix file parsing, update tests

iuhiah committed:
Merge branch 'A-Varargs'

iuhiah committed:
Fix checkstyle issues

iuhiah committed:
Merge branch 'A-CheckStyle'

iuhiah committed:
Change GUI layout, unstable build

iuhiah committed:
Add assertions for null checks

iuhiah committed:
Fix code quality issues

iuhiah committed:
Refactor TaskList methods to use streams

iuhiah committed:
Merge pull request #2 from iuhiah:branch-A-Assertions

Add assertions for null checks

iuhiah committed:
Merge pull request #3 from iuhiah:branch-A-Streams

Add streams to TaskList

iuhiah committed:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

iuhiah committed:
Merge pull request #1 from iuhiah:branch-A-CodeQuality

Fix code quality issues

Code quality issues in the project.

- Split up longer methods in Bibo.java
- Extract duplicated lines into a method
- Reuse the methods in Command.java and TaskList.java
- Clean up code to remove unnecessary comments and methods

This will improve the readability and maintainability of the code.

iuhiah committed:
Fix command responses

iuhiah committed:
Add note management functionality, fix code quality issues

No note management functionality.

- Add Note and Notes classes
- Add relevant exceptions
- Refactor old code to handle method conflicts

Methods and classes must align with new naming structure from adding notes.

iuhiah committed:
Add GitHub Actions workflow for Java CI with Gradle build

iuhiah committed:
Merge branch 'master' into A-BetterGui

iuhiah committed:
Add UI styling

iuhiah committed:
Fix styling and update gitignore to untrack data folders

iuhiah committed:
Add send button

iuhiah committed:
Add more error handling

iuhiah committed:
Fix code quality issues

iuhiah committed:
Add user guide and fix bugs

iuhiah committed:
Update user guide

iuhiah committed:
Rename product demo photo to Ui.png

iuhiah committed:
Update user guide with accurate expect output

iuhiah committed:
Fix code quality issues and reorder platform to check workflow

iuhiah committed:
Fix file creation methods in Storage class

iuhiah committed:
Fix storage tests
